src/InterceptedHttpClient.php

```diff
@@ -0,0 +1,29 @@
+<?php
+
+namespace Amp\Http\Client;
+
+use Amp\CancellationToken;
+use Amp\NullCancellationToken;
+use Amp\Promise;
+
+final class InterceptedHttpClient implements HttpClient
```

Comment:
Should this class be an internal class?

Reply:
I don't see a reason to make it internal.

Reply:
Reasoning was to ensure users use `HttpClientBuilder` instead of this class to add application interceptors.

Reply:
If you want to build another builder, that's fine and you can reuse `InterceptedHttpClient` then.